app/src/ui/delete-branch/delete-branch-dialog.tsx

```diff
@@ -63,15 +120,60 @@ export class DeleteBranch extends React.Component<
     )
   }
 
+  private renderLocalUnmergedWarning(unmergedCommits: number) {
+    if (this.state.existsOnRemote || this.state.localUnmergedCommits <= 0) {
+      return null
+    }
+
+    return (
+      <div>
+        <strong>
+          Warning: This branch has {unmergedCommits} unmerged{' '}
+          {unmergedCommits === 1 ? 'commit' : 'commits'}.
+        </strong>
+      </div>
+    )
+  }
+
+  private renderUnpushedCommitsWarning(unpushedCommits: number) {
+    if (unpushedCommits <= 0) {
+      return null
+    }
+
+    return (
+      <div>
+        <strong>
+          Warning: This branch has {unpushedCommits} unpushed{' '}
+          {unpushedCommits === 1 ? 'commit' : 'commits'}.
+        </strong>
+      </div>
+    )
+  }
+
+  private renderRemoteUnmergedCommitsWarning(unmergedCommits: number) {
+    if (unmergedCommits <= 0) {
+      return null
+    }
+
+    return (
+      <div>
+        <strong>
+          Warning: <Ref>{this.props.branch.upstream}</Ref> has {unmergedCommits}{' '}
+          unmerged {unmergedCommits === 1 ? 'commit' : 'commits'}.
+        </strong>
+      </div>
+    )
+  }
+
   private renderDeleteOnRemote() {
-    if (this.props.branch.upstreamRemoteName && this.props.existsOnRemote) {
+    if (this.props.branch.upstreamRemoteName && this.state.existsOnRemote) {
       return (
         <div>
-          <p id="delete-branch-confirmation-message-remote">
-            <strong>
-              The branch also exists on the remote, do you wish to delete it
-              there as well?
-            </strong>
+          <p>
```

Comment:
```suggestion
          <p id="delete-branch-confirmation-message-remote">
```
Need to keep for the alert dialog to announce this.